sim/druid/starsurge.go

```diff
@@ -0,0 +1,59 @@
+package druid
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+)
+
+func (druid *Druid) registerStarsurgeSpell() {
+	solarMetric := druid.NewSolarEnergyMetric(core.ActionID{SpellID: 78674})
+	lunarMetric := druid.NewLunarEnergyMetrics(core.ActionID{SpellID: 78674})
+
+	druid.Starsurge = druid.RegisterSpell(Humanoid|Moonkin, core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 78674},
+		SpellSchool:    core.SpellSchoolArcane | core.SpellSchoolNature,
+		ProcMask:       core.ProcMaskSpellDamage,
+		ClassSpellMask: DruidSpellStarsurge | DruidArcaneSpells | DruidNatureSpells,
+		Flags:          SpellFlagOmenTrigger | core.SpellFlagAPL,
+
+		DamageMultiplier:         1,
+		DamageMultiplierAdditive: 1,
+		CritMultiplier:           druid.DefaultSpellCritMultiplier(),
+		ManaCost: core.ManaCostOptions{
+			BaseCost:   0.17,
+			Multiplier: 1,
+		},
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD:      core.GCDDefault,
+				CastTime: time.Second * 2,
+			},
+			CD: core.Cooldown{
+				Timer:    druid.NewTimer(),
+				Duration: time.Second * 15,
+			},
+		},
+
+		ThreatMultiplier: 1,
+
+		BonusCoefficient: 1.228,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			min, max := core.CalcScalingSpellEffectVarianceMinMax(proto.Class_ClassDruid, 1.228, 0.32)
+			baseDamage := sim.Roll(min, max)
+			result := spell.CalcDamage(sim, target, baseDamage, spell.OutcomeMagicHitAndCrit)
+
+			if result.Landed() {
+				druid.AddEclipseEnergy(15, SolarEnergy, sim, solarMetric)
```

Comment:
I made it like that on purpose as the first version, because you don't actually want to cast SS while neutral and I was going to implement it later. Thanks for this :)

Reply:
done.